go/vt/vtgate/evalengine/api_literal.go

```diff
@@ -201,35 +202,38 @@ func NewLiteralBinaryFromBit(val []byte) (*Literal, error) {
 // NewBindVar returns a bind variable
 func NewBindVar(key string, typ Type) *BindVariable {
 	return &BindVariable{
-		Key:       key,
-		Type:      typ.Type,
-		Collation: defaultCoercionCollation(typ.Coll),
+		Key:               key,
+		Type:              typ.Type,
+		Collation:         typ.Coll,
+		dynamicTypeOffset: -1,
 	}
 }
 
 // NewBindVarTuple returns a bind variable containing a tuple
-func NewBindVarTuple(key string, col collations.ID) *BindVariable {
+func NewBindVarTuple(key string, coll collations.ID) *BindVariable {
 	return &BindVariable{
 		Key:       key,
 		Type:      sqltypes.Tuple,
-		Collation: defaultCoercionCollation(col),
+		Collation: coll,
 	}
 }
 
 // NewColumn returns a column expression
-func NewColumn(offset int, typ Type) *Column {
+func NewColumn(offset int, typ Type, original sqlparser.Expr) *Column {
 	return &Column{
-		Offset:    offset,
-		Type:      typ.Type,
-		Collation: defaultCoercionCollation(typ.Coll),
+		Offset:            offset,
+		Type:              typ.Type,
+		Collation:         defaultCoercionCollation(typ.Coll),
+		Original:          original,
+		dynamicTypeOffset: -1,
 	}
 }
 
 // NewTupleExpr returns a tuple expression
 func NewTupleExpr(exprs ...Expr) TupleExpr {
 	tupleExpr := make(TupleExpr, 0, len(exprs))
 	for _, f := range exprs {
-		tupleExpr = append(tupleExpr, f)
+		tupleExpr = append(tupleExpr, f.(IR))
```

Comment:
Assuming this can never be anything but an `IR` type?

Reply:
Yeah I should lift that into the signature TBH.

Reply:
Fixed